File: from-1801-to-1900/1823-find-the-winner-of-the-circular-game/solution.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        q = deque([i for i in range(1, n+1)])
        cnt = 0
        while q:
            player = q.popleft()
            cnt += 1
            if cnt != k:
                q.append(player)
            else:
                cnt = 0

        return player

File: from-1801-to-1900/1823-find-the-winner-of-the-circular-game/solution.go
func findTheWinner(n int, k int) int {
    players := make([]int, n)
    var player int
    for i := 0; i < n; i ++ {
        players[i] = i + 1
    }
    count := 0
    for len(players) > 0 {
        player = players[0]
        players = players[1:]
        count += 1
        if count == k {
            count = 0
        } else {
            players = append(players, player)
        }
    }
    return player
}